Creating About us section including progress bar component

alaasaedy opened the pull request.

alaasaedy committed:
Creating About us section including progress bar component

alaasaedy committed:
adjust paragraph margin,  line height and color to align with design

MAlngaawy commented on frontend/src/App.js:
```diff
@@ -12,6 +13,11 @@ export default class App extends React.Component {
+    return (
+      <div>
+        <AboutUs />
+        <Prototype />
```
please recommend the prototype component

MAlngaawy commented on frontend/src/components/AboutUsSection/AboutUs.jsx:
```diff
@@ -0,0 +1,53 @@
+import './style.css';
+
+function AboutUs() {
+  const progressBarData = [
```
bring the progress bar data from the backend and use useEffect

alaasaedy replied:
will work for it

MAlngaawy commented on frontend/src/components/AboutUsSection/AboutUs.jsx:
```diff
@@ -0,0 +1,53 @@
+import React from 'react';
+import { Button } from '../shared/button';
+import ProgressBar from '../ProgressBar/ProgressBar';
+import './style.css';
```
always avoid using any CSS file

MAlngaawy commented on frontend/src/components/AboutUsSection/AboutUs.jsx:
```diff
@@ -0,0 +1,53 @@
+        <Button
+          bgColor="c100"
+          rounded={true}
+          customClassNames="my-4 block w-56 h-16 absolute button__position"
```
try to centralize it by anything than absolute position 
try mx-auto or any other thing doesn't require a custom CSS file

MAlngaawy commented on frontend/src/components/AboutUsSection/AboutUs.jsx:
```diff
@@ -0,0 +1,53 @@
+          rounded={true}
+          customClassNames="my-4 block w-56 h-16 absolute button__position"
+        >
+          GET IN TOUCH
```
the reusable button have class uppercase son no need to add it as uppercase

MAlngaawy commented on frontend/src/components/AboutUsSection/AboutUs.jsx:
```diff
@@ -0,0 +1,53 @@
+          quis sit amet tortor. Proin ornare, lorem ut pellentesque maximus, ex
+          leo viverra lorem, ac dapibus neque tellus id dolor. Curabitur luctus
+          tortor eget dolor ultrices fermentum.
+        </p>
```
please bring the paragraph and the title also from the API

MAlngaawy commented on frontend/src/components/AboutUsSection/AboutUs.jsx:
```diff
@@ -0,0 +1,53 @@
+        </p>
+        {progressBarData.map((item, idx) => (
+          <ProgressBar
+            key={idx}
```
try using another key than the index
For example, from the API too

MAlngaawy commented on frontend/src/components/AboutUsSection/AboutUs.jsx:
```diff
@@ -0,0 +1,53 @@
+          <ProgressBar
+            key={idx}
+            bgcolor={item.bgcolor}
+            completed={item.completed}
```
**progress** will be more realistic than the completed word.

alaasaedy replied:
Do you mean the component name ?

MAlngaawy commented on frontend/src/components/ProgressBar/ProgressBar.jsx:
```diff
@@ -0,0 +1,34 @@
+import React from 'react';
+
+const ProgressBar = props => {
+  const { bgcolor, completed, label } = props;
```
please, try to destructure it at the argument, instead of props add `({ bgcolor, completed, label }) => `
**and don't forget the icons**

MAlngaawy commented on frontend/src/components/AboutUsSection/AboutUs.jsx:
```diff
@@ -0,0 +1,53 @@
+        </Button>
+      </div>
+      <div>
+        <h6 className="text-xxlg text-c200 mt-1">Read About My Life Story</h6>
```
consider the font family `PT serif`

MAlngaawy reviewed (changes requested):
nice work Alaa, but consider these changes, please.

alaasaedy committed:
using inline styling,  centerlize the btn component without using absolute positioning,  change btn label to lower case and let the global styling for the btn do the work.
change mapped items key until I can use the data coming from API.
using destructure feature.
change paragraph title font family

alaasaedy committed:
adding data from DB

alaasaedy committed:
Adding section data dynamically

amagdy46 commented on frontend/src/components/AboutUsSection/AboutUs.jsx:
```diff
@@ -0,0 +1,70 @@
+import { API } from '../../modules/apis';
+
+const AboutUs = () => {
+  const [sectionData, setSectionData] = useState({});
```
don't initialize the sate with an empty object.

just do useState();

this is what resulted in the unexpected behavior

alaasaedy committed:
adding icon prop to progressbar component and resize it to fit side by side with the title.
change naming to fetched data and add id, icon to the DB data to use it in the frontend

alaasaedy committed:
change props naming

alaasaedy committed:
Merge branch 'master' of https://github.com/m3ntorship/personal-resume into abour-us-section

alaasaedy committed:
add heading component with needed props to the section

amagdy46 reviewed (approved):
LGTM!

MAlngaawy committed:
Merge branch 'master' of https://github.com/m3ntorship/personal-resume into abour-us-section

MAlngaawy committed:
add margin to the section
